packages/integration-tests/src/tests/api/hooks.test.ts

```diff
@@ -8,8 +8,13 @@ import { enableAllPasswordSignInMethods } from '#src/helpers/sign-in-experience.
 import { generateNewUser, generateNewUserProfile } from '#src/helpers/user.js';
 import { waitFor } from '#src/utils.js';
 
-const createPayload = (event: HookEvent, url = 'not_work_url'): Partial<Hook> => ({
-  event,
+type CreateHookPayload = Pick<Hook, 'name' | 'events'> & {
+  config: HookConfig;
+};
+
+const createPayload = (event: HookEvent, url = 'not_work_url'): CreateHookPayload => ({
+  name: 'hook_name',
+  events: [event],
```

Comment:
Should we keep the original test case using the event as input payload instead of the new events?

Reply:
New tests added:
- create a hook by the original payload format
- trigger a hook created by the original payload format